extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanServiceBindingConverter.java

```diff
@@ -0,0 +1,102 @@
+package io.quarkus.infinispan.client.runtime;
+
+import static java.lang.String.format;
+import static org.apache.sshd.common.util.GenericUtils.isBlank;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBinding;
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBindingConfigSource;
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBindingConverter;
+
+/**
+ * ServiceBindingConverter for Infinispan to support SBO (Service Binding Operator) in Quarkus.
+ *
+ * <br>
+ * <br>
+ * Following individual properties are supported to make the connection:
+ * <ul>
+ * <li>username</li>
+ * <li>password</li>
+ * <li>useauth</li>
+ * <li>server (<i>server information can be provided in this property separated by : sign</i>, e.g. localhost:11222</li>
+ * </ul>
+ * <i>Other than server all other properties are optional</i>
+ *
+ * <br>
+ * <br>
+ * The Quarkus properties set by this class are:
+ * <ul>
+ * <li>quarkus.infinispan-client.server-list</li>
+ * <li>quarkus.infinispan-client.use-auth (<i>if useauth is provided</i>)</li>
+ * <li>quarkus.infinispan-client.auth-username (<i>if username is provided</i>)</li>
+ * <li>quarkus.infinispan-client.auth-password (<i>if password is provided</i>)</li>
+ * </ul>
+ */
+public class InfinispanServiceBindingConverter implements ServiceBindingConverter {
+
+    private static final Logger LOGGER = Logger.getLogger(InfinispanServiceBindingConverter.class);
+    private static final String BINDING_TYPE = "infinispan";
+    public static final String BINDING_CONFIG_SOURCE_NAME = BINDING_TYPE + "-k8s-service-binding-source";
+
+    // Connection properties
+    public static final String INFINISPAN_SERVER_LIST = "serverlist";
```

Comment:
I think we can just use the `uri` binding key now and use it's value to set the `server-list` property. The `HotRodURI` class can be used to parse the string and extract the addresses of the server.

Of course we can sitll add the `serverlist` binding, but `uri` is one of the well known keys proposed in the [spec](https://servicebinding.io/spec/core/1.0.0/). It also means it's one less thing to deprecate/communicate when deprecating `server-list` in favour of `uri` in the application.properties.

The `HotRodURI` class also extracts the user/password combo, so we can set those properties from the `uri` binding if present.

Reply:
I will use url and pass this PR after the deprecation, but I did it before so that's why I used serverList
In any case, my issue is I don't know how to validate this

Reply:
You can test by setting the quarkus.kubernetes-service-binding.root=/tmp property, you then need to create a file structure containing the servicebinding files, e.g:

```
/tmp/infinispan
├── username
├── password
└── type
```
Where each file contains the value associated with the filename.

If the ServiceBinding converter is working as expected, then the Infinispan client's properties should be configured with the values in those files.

Reply:
I did that, but in which environment did you do that ?

Reply:
> I did that, but in which environment did you do that ?

Just on my local dev machine.

Looking at the PR, I think I know why it didn't work.

You need to define the `io.quarkus.kubernetes.service.binding.runtime.ServiceBindingConverter` implementation as a Service in `src/resources/META-INF/services/io.quarkus.kubernetes.service.binding.runtime.ServiceBindingConverter`

https://github.com/gingersnap-project/db-syncer/blob/ae9ba3643b59d2dab87767983cad83d5fd290f96/src/main/resources/META-INF/services/io.quarkus.kubernetes.service.binding.runtime.ServiceBindingConverter#L3-L2